src/processors/default/createERC721NFTsFromTransfersProcessor.ts

```diff
@@ -31,38 +32,60 @@ const processorFunction = (contracts: NftFactory[]) =>
+      const contractAddress = formatAddress(contract.id);
       const tokenId = BigInt((item.args!.tokenId as BigNumber).toString());
-      const newMint = item.args!.from === ETHEREUM_NULL_ADDRESS;
+      const nftId = contractType.buildNFTId(contractAddress, tokenId);
+
       transfers.push({
-        id: `${CHAIN}/${item.blockNumber}/${item.logIndex}`,
-        contractAddress: formatAddress(contract.id),
-        from: item.args!.from,
-        to: item.args!.to,
+        id: ethereumTransferId(item.blockNumber, item.logIndex),
+        contractAddress: contractAddress,
+        from: fromAddress,
+        to: toAddress,
         tokenId,
         createdAtEthereumBlockNumber: '' + item.blockNumber,
-        nftId: contractType.buildNFTId(contract.id, tokenId), 
+        nftId: nftId,
         transactionHash: item.transactionHash
       });
-      if (!newMint) {
-        updates.push({
-          id: contractType.buildNFTId(contract.id, tokenId),
-          owner: item.args!.to
+
+      if (!newMint(fromAddress)) {
+        updatedNFTs.push({
+          id: nftId,
+          owner: toAddress,
+          burned: burned(toAddress)
         })
-        return undefined;
+        nftsCollectorsChanges.push({ nftId: nftId, collectorId: fromAddress, amount: -1 })
+        if (!burned(toAddress)) {
+          nftsCollectorsChanges.push({ nftId: nftId, collectorId: toAddress, amount: 1 })
+        }
+      } else {
+        newNFTs.push({
+          id: nftId,
+          createdAtEthereumBlockNumber: '' + item.blockNumber,
+          contractAddress: contractAddress,
+          tokenId,
+          platformId: contract.platformId,
+          owner: toAddress,
+          approved: contract.autoApprove
+        });
+        nftsCollectorsChanges.push({ nftId: nftId, collectorId: toAddress, amount: 1 })
       }
-      newNFTs.push({
-        id: contractType.buildNFTId(contract.id, tokenId),
-        createdAtEthereumBlockNumber: '' + item.blockNumber,
-        contractAddress: formatAddress(contract.id),
-        tokenId,
-        platformId: contract.platformId,
-        owner: item.args!.to,
-        approved: contract.autoApprove
-      });
     });
 
+    const allCollectors = nftsCollectorsChanges.map<Collector>(({ collectorId }) => { return { id: collectorId } } );
+    const idPairs = _.uniq(nftsCollectorsChanges.map( (e) => [e.nftId, e.collectorId] ))
+    const existingNftsCollectors = await clients.db.getRecords<NFTsCollectors>(Table.nftsCollectors,
+      [ ['whereIn', [ ['nftId', 'collectorId'], idPairs ] ] ]
+    );
+
+    const allNftsCollectorsChanges = nftsCollectorsChanges.concat(existingNftsCollectors)
+    const updatedNftsCollectors = consolidate(allNftsCollectorsChanges)
+
     await clients.db.insert(Table.nfts, newNFTs.filter(n => !!n), { ignoreConflict: 'id' });
-    await clients.db.update(Table.nfts, updates);
+    await clients.db.upsert(Table.nfts, updatedNFTs);
```

Comment:
Still don't follow the scenario in which it's possible for updatedNFTs to include an nft that does not yet exist, so don't understand the need for an upsert here. Can you clarify?

Reply:
The culprit was an [off by one error](https://www.martinfowler.com/bliki/TwoHardThings.html) with a starting block for the ZoraOriginal contract. Resolved!